gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt

```diff
@@ -25,6 +26,25 @@ class ComposeCompilerKotlinSupportPlugin : KotlinCompilerPluginSupportPlugin {
                     ComposeCompilerCompatibility.compilerVersionFor(target.getKotlinPluginVersion())
             }
         }
+        target.afterEvaluate {
+            warnAboutJetpackComposeCompilerUsageForNonJvm(it)
```

Comment:
This will be shown multiple times, as for the error (https://github.com/JetBrains/compose-multiplatform/issues/2866).

The fix is probably using some global AtomicBoolean, but I am not sure where to place it (Gradle can cache confiugurations).

So, i just mentioned it to not forget to fix it in the future.

Reply:
@eymar @igordmn 

If there is a cache entry for a particular set of scripts, plugins, build arguments and environment variables, configuration phase will be skipped (including `afterEvaluate` and all code under `Plugin.apply`)

One way to avoid that and show a warning just once is to use a build service.

```
import org.gradle.api.Plugin
import org.gradle.api.Project
import org.gradle.api.logging.Logging
import org.gradle.api.provider.Provider
import org.gradle.api.provider.SetProperty
import org.gradle.api.services.BuildService
import org.gradle.api.services.BuildServiceParameters
import org.gradle.api.services.BuildServiceRegistry
import org.gradle.build.event.BuildEventsListenerRegistry
import org.gradle.tooling.events.FinishEvent
import org.gradle.tooling.events.OperationCompletionListener
import javax.inject.Inject

// The service implements OperationCompletionListener just so we Gradle would use the service
// even if the service is not used by any task or transformation
abstract class ComposeMultiplatformBuildService : BuildService<ComposeMultiplatformBuildService.Parameters>, OperationCompletionListener, AutoCloseable {
    interface Parameters : BuildServiceParameters {
        val postBuildWarnings: SetProperty<String>
    }

    private val log = Logging.getLogger(this.javaClass)

    override fun close() {
        parameters.postBuildWarnings.get().forEach {
            log.error(it)
        }
    }

    override fun onFinish(event: FinishEvent) {}

    companion object {
        fun configure(project: Project, fn: Parameters.() -> Unit): Provider<ComposeMultiplatformBuildService> =
            project.gradle.sharedServices.registerOrConfigure<Parameters, ComposeMultiplatformBuildService> {
                fn()
            }
    }
}

inline fun <reified P : BuildServiceParameters, reified S : BuildService<P>> BuildServiceRegistry.registerOrConfigure(
    crossinline fn: P.() -> Unit
): Provider<S> {
    val serviceClass = S::class.java
    val serviceFqName = serviceClass.canonicalName
    val existingService = registrations.findByName(serviceFqName)
        ?.apply { (parameters as? P)?.fn() }
        ?.service
    return (existingService as? Provider<S>)
        ?: registerIfAbsent(serviceFqName, serviceClass) {
            parameters.fn()
        }
}
```

The service first be registered like this:
```
abstract class ComposePlugin : Plugin<Project> {
    @Inject
    abstract fun getEventsListenerRegistry(): BuildEventsListenerRegistry

    override fun apply(project: Project) {
        val service = ComposeMultiplatformBuildService.configure(project) {}
        // we add the service to the BuildEventsListenerRegistry, so
        // Gradle uses it even if the service is not used by tasks
        getEventsListenerRegistry().onTaskCompletion(service)
    }
}
```

Then, a warning can be added like this:
```
ComposeMultiplatformBuildService.configure(project) {
    postBuildWarnings.add("A warning that will be shown once per build!")
}   
```